### ISSUE TITLE
'ascii' codec can't encode character '\xe7' in position [...]: ordinal not in range(128)

### ISSUE THREAD

**Issue by marcelpaulo** [jarun/ddgr#52]
Have just installed from master (141f5b7e7e60d3851596787c3c5bc5458bcc35d9), and got this error when completing a word with non-ascii characters:

```
+paulo@monk:~/tmp$ ddgr --complete traduç
Traceback (most recent call last):
  File "/usr/local/bin/ddgr", line 1807, in <module>
    main()
  File "/usr/local/bin/ddgr", line 1737, in main
    completer_run(opts.complete)
  File "/usr/local/bin/ddgr", line 1670, in completer_run
    completions = completer_fetch_completions('+'.join(prefix.split()))
  File "/usr/local/bin/ddgr", line 1662, in completer_fetch_completions
    resp = urllib.request.urlopen(api_url, timeout=3)
  File "/usr/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.6/urllib/request.py", line 526, in open
    response = self._open(req, data)
  File "/usr/lib/python3.6/urllib/request.py", line 544, in _open
    '_open', req)
  File "/usr/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.6/urllib/request.py", line 1361, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/usr/lib/python3.6/urllib/request.py", line 1318, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/usr/lib/python3.6/http/client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1250, in _send_request
    self.putrequest(method, url, **skips)
  File "/usr/lib/python3.6/http/client.py", line 1117, in putrequest
    self._output(request.encode('ascii'))
UnicodeEncodeError: 'ascii' codec can't encode character '\xe7' in position 16: ordinal not in range(128)
```

Running python 3.6.3 on Xubuntu 17.10 with xfce4-terminal 0.8.6 + tmux (b5c0b2ca), and these locale:

```
+paulo@monk:~/tmp$ locale
LANG=en_US.UTF-8
LANGUAGE=en_US
LC_CTYPE=pt_BR.UTF-8
LC_NUMERIC=pt_BR.UTF-8
LC_TIME=pt_BR.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=pt_BR.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=pt_BR.UTF-8
LC_NAME=pt_BR.UTF-8
LC_ADDRESS=pt_BR.UTF-8
LC_TELEPHONE=pt_BR.UTF-8
LC_MEASUREMENT=pt_BR.UTF-8
LC_IDENTIFICATION=pt_BR.UTF-8
LC_ALL=
```

**Comment by marcelpaulo** [comment]
Have just checked now, and `googler` doesn't have this issue:

```
+paulo@monk:~/tmp$ googler --complete traduç
tradução
tradução bella ciao
tradução havana
traduções gold
tradução perfect
tradução gta san andreas
tradução skyrim
tradução doki doki
tradução downtown
tradução juramentada curitiba
```

**Comment by marcelpaulo** [comment]
@jarun, you were right, DDG seems to return UTF-8:

```
+paulo@monk:~/tmp$ curl -s -I -A 'Python-urllib/3.6' 'https://duckduckgo.com/ac/?q=hello&kl=wt-wt'
HTTP/1.1 200 OK
Server: nginx
Date: Tue, 10 Apr 2018 06:44:52 GMT
Content-Type: application/javascript; charset=UTF-8
Connection: keep-alive
Strict-Transport-Security: max-age=31536000
X-Frame-Options: SAMEORIGIN
Content-Security-Policy: default-src https: blob: data: 'unsafe-inline' 'unsafe-eval'; frame-ancestors 'self'
X-XSS-Protection: 1;mode=block
X-Content-Type-Options: nosniff
Referrer-Policy: origin
Expect-CT: max-age=0
Expires: Tue, 10 Apr 2018 06:44:51 GMT
Cache-Control: no-cache
Content-Disposition: attachment; filename="ac.json"
X-DuckDuckGo-Privacy-Random-AC: 4Sn0Yc9TGYwuTgrYfkAXzj29l6cAgQP8TEaveiNmQSSwN06OJJ1jSAn9hItSGlR8HEVpfyiBisbaxqg5EuOwl4gwWgfFxqtK723VsogR4ZCqgLnvSGV9zixzIoK8g4XDydpHjjGH6hg4uC9uctGVK7YdOcCEamV8DS9aPgAZmdqajMb7H4hyU2XDelgNpvs17hvaGS0099dR7SRTxl4E5VvJimZqTfwH5wmMpx5Vibp4zuRc2FTjcT6ryCd4g7TyWmKxnGIuweEOSD6j6dtX1QGPGAYdxzdOL57FwcjxHFmi0DVisEeU4bhPmasQid4ioinqu49u4g2oB2LgLR3f5o2CDUMZ42WO
```

**Comment by marcelpaulo** [comment]
I get the impression here the problem lies in the way `api_url` is encoded. Comparing `googler` with `ddgr`:

From `googler`:
```python
    api_url = ('https://www.google.com/complete/search?client=psy-ab&q=%s' %
               urllib.parse.quote(prefix, safe=''))
```

From `ddgr`:
```python
    api_url = ('https://duckduckgo.com/ac/?q=%s&kl=wt-wt' % prefix)
```

I'll do some more digging.

**Comment by jarun** [comment]
> I'll do some more digging.

Thanks! It would be great if you can submit a patch.

**Comment by marcelpaulo** [comment]
> It would be great if you can submit a patch

Working on it this very moment ...

**Comment by marcelpaulo** [comment]
After #53:

```
:paulo@monk:~/tmp$ ddgr --complete traduç
tradução
tradução de ingles para portugues
traduções
tradução google
tradução ingles
traduçao de musicas
tradução francês português
tradução despacito
tradução ingles portugues
tradução em portugues
```

Fixed by #53.
